File: lettercase-percentage-ratio.py
import sys

def solve(problem):
    lowercase = sum(map(lambda x: 1 if x.islower() else 0, list(problem)))
    uppercase = sum(map(lambda x: 1 if x.isupper() else 0, list(problem)))
    output = 'lowercase: {:.2f} uppercase: {:.2f}'
    length = len(problem)
    return output.format((lowercase*100 / length), (uppercase*100 / length))

def main():
    with open(sys.argv[1]) as input_file:
        for line in input_file:
            print(solve(line.strip()))

if __name__ == '__main__':
    main()

File: lettercase-percentage-ratio.lua
function solve(line)
    lowercase = 0
    for _ in line:gmatch('[a-z]') do lowercase = lowercase + 1 end
    uppercase = 0
    for _ in line:gmatch('[A-Z]') do uppercase = uppercase + 1 end
    
    print('lowercase: %f uppercase: %f', lowercase * 100 / line:len(), (uppercase * 100 / line:len()))
    -- return ('lowercase: ' .. lowercase .. ' uppercase: ' .. uppercase .. '\n')

end

for line in io.lines(arg[1]) do
    solve(line)
end
